Attempting to install an older version of `wasm-tools` fails (even version 1.234.0)

# Problem

When trying to install an older version of wasm-tools, you get a compilation error:

```
   ....
   Compiling is_executable v1.0.4
   Compiling bytesize v2.0.1
error[E0432]: unresolved import `wast::component::ComponentKind`
  --> /Users/evgilber/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasm-tools-1.234.0/src/bin/wasm-tools/wast.rs:14:34
   |
14 | use wast::component::{Component, ComponentKind};
   |                                  ^^^^^^^^^^^^^
   |                                  |
   |                                  no `ComponentKind` in `component`
   |                                  help: a similar name exists in the module: `Component`

error[E0026]: struct `wast::component::Component` does not have a field named `kind`
   --> /Users/evgilber/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasm-tools-1.234.0/src/bin/wasm-tools/wast.rs:693:13
    |
693 |             kind: ComponentKind::Binary(_),
    |             ^^^^ struct `wast::component::Component` does not have this field

Some errors have detailed explanations: E0026, E0432.
For more information about an error, try `rustc --explain E0026`.
error: could not compile `wasm-tools` (bin "wasm-tools") due to 2 previous errors
error: failed to compile `wasm-tools v1.234.0`, intermediate artifacts can be found at `/var/folders/fv/k9bn41tx2wb8_xl37sf30b_c0000gq/T/cargo-installqYEDWi`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

# To Reproduce

Simply run: `cargo install wasm-tools --version 1.234.0`

# Possible fix??

It might be that the newest version of the `wast` crate changed this structure of a `Component`, it might be fixed by patching your previous releases by adding a `=` to the beginning of your `wast` version in `Cargo.toml`, e.g.:
 `wast = { version = "=235.0.0", path = "crates/wast", default-features = false }`

(Rust will automatically pull patch releases, so they may have changed this structure in a patched release automatically pulled by the Rust compiler.)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -201,7 +201,7 @@ cpp_demangle = { version = "0.4.0", optional = true }
 rustc-demangle = { version = "0.1.21", optional = true }
 
 # Dependencies of `component`
-wast = { workspace = true, optional = true, features = ['wasm-module'] }
+wast = { workspace = true, optional = true, features = ['wasm-module', 'component-model'] }
 wit-component = { workspace = true, optional = true, features = ['dummy-module', 'wat', 'semver-check'] }
 wit-encoder = { workspace = true, optional = true }
 wit-parser = { workspace = true, optional = true, features = ['decoding', 'wat', 'serde'] }

